### DISCUSSION TITLE
変更したimageviewとlabelのデータをアプリを再起動しても変更した情報が残っているようにしたい

### DISCUSSION THREAD

**Discussion by ghost** [lit-iphone-course/lit-iphone-course#34] [Question-original] [answered]
(1) 実現したいこと
変更したimageviewとlabelのデータをアプリを再起動しても変更した情報が残っているようにしたい
(2) 実現するために自分が試した内容とその結果
Realmにimage view とLabelを保存＞エラー
(3) 調べた内容や資料
教科書
(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
`let data = Grow()
        data.level =  LevelLabel.text!
        
        try! realm.write {
            realm.Grow()`
(5) GitHubのリンク
https://github.com/j-fuuuu/frog.git

**Comment by txxxxc** [comment]
Realmに保存しているデータをアップデートするには`realm.write{}`の中でそのデータを書き込む必要があります！
なので、以下のような流れで処理を実装するといい感じだと思います！
1. realmに保存されているキャラクターのデータを取り出す（なければ新規作成する）
2. キャラクターの画像データが変わる(レベルが上った)場合取り出したデータに`realm.write{}`で画像のファイル名を書き込む
(ちなみにRealmにUIImageViewやLabelは保存できないので、その代わりに画像名やLabelに表示されているテキストを保存して、それを毎回UIに反映させるという流れが一般的です！)

やりたい事的にはSchoolのRealmの教科書とほぼ一緒だと思うので、もう一度見返して見ると良さそうです！
参考にしてみてくださいー！

**Reply by ghost** [reply]
imageviewの画像ファイル名をRealmに保存するというところができていない・分からず詰まっています。
詳しく教えていただけると嬉しいです

**Reply by txxxxc** [reply]
多分アプリの仕様的にアプリ内に予め要されている画像を切り替える感じだと思う(違ってたら教えて下さい！)ので、画像ファイル名をStringで保存しておいてそれを使ってUIImageを初期化すればやりたいことが達成できそうだなと思ってます！ファイル名はStringなので、Realmの教科書でメモ(String)を保存する時と全く一緒です！

例としてはだいたいこんな感じです！

```swift
let data =  realm.objects(Data.self).first
guard let data = data else { return }
uiImage = UIImage(named: data.imageName)
uiImageView.image = uiImage
```
なにか不明な点があったら何でも言ってください！(そもそも聞きたい部分と違う！みたいなやつがあればいつでも言ってください！）

**Reply by ghost** [reply]
https://github.com/j-fuuuu/frognote24.git

すみません。githubに一度あげると次にPushした時に、「The local repository is out of date.」と表示されてできないので新しく挙げています。

以下このスレッドの質問

`guard let data = data else { return }`

のところでエラーが出るのですがどうすれば良いのでしょうか

**Reply by txxxxc** [reply]
GitHubのやつに関しては一回これを試してみて欲しい！！
https://qiita.com/shimamura_io/items/5e65843fae788f46032c

エラーの方はどんなエラー出てるかコピペで教えて欲しいです！！

**Reply by ghost** [reply]
githubの方は記事のやり方をやっても同じエラー文が出ます

characteImageView = UIImage(named: data."frog1")
＞Expected member name following '.'
characteImageView.image = uiImage
＞Cannot find 'uiImage' in scope

**Comment by yukiny0811** [comment, marked as answer]
countをuserdefaultsで保存しておいて、画面読み込み時に読み込んだcountの値に応じで画像を切り替える実装で解決！
